### PR TITLE
Fix english doc api, invloves the op of retinanet_target_assign, sigmoid_focal_loss and retinanet_detection_output.

### PR DESCRIPTION
**Pull request by FlyingQianMM**
test=document_fix
test=develop
Fix english doc api, invloves the op of retinanet_target_assign, sigmoid_focal_loss and retinanet_detection_output.

### CODE REVIEW

**Comment by FlyingQianMM** [comment]
e8652ccfce1064cce0d2bcc7275c8c80
9301a0a0e9754b1ab3751dd5d53445fb

**Comment by FlyingQianMM** [comment]
2b01eecd7b4feaf004bfd012c9a223f3
90571f64cea95a3713e10aa9a6eea248

**Comment by FlyingQianMM** [comment]
0b38dc1b4414b8a14d1dcf5a2ce847c2
8ff9d6a1a2badddef820931bf941c2fd

File: python/paddle/fluid/layers/detection.py [hunk 1]
```diff
@@ -73,100 +73,166 @@ def retinanet_target_assign(bbox_pred,
                             positive_overlap=0.5,
                             negative_overlap=0.4):
     """
-    **Target Assign Layer for Retinanet .**
+    **Notice: this OP supports CPU mode only.**
```
Comment (heavengate) [inline python/paddle/fluid/layers/detection.py]: 只支持CPU不用单独给出，容易让用户误解没法用GPU训练

File: python/paddle/fluid/layers/detection.py [hunk 2]
```diff
@@ -2635,78 +2714,109 @@ def retinanet_detection_output(bboxes,
...
-                                                    im_info=im_info,
-                                                    score_threshold=0.05,
-                                                    nms_top_k=1000,
-                                                    keep_top_k=100,
-                                                    nms_threshold=0.3,
-                                                    nms_eta=1.)
+           import paddle.fluid as fluid
+
+           bboxes_low = fluid.data(
+               name='bboxes_low', shape=[1, 44, 4], dtype='float32')
```
Comment (heavengate) [inline python/paddle/fluid/layers/detection.py]: shape前面不加None/-1的话会不会导致维度错误

File: python/paddle/fluid/layers/detection.py (hunk 1 above)
Comment (FlyingQianMM) [reply python/paddle/fluid/layers/detection.py]: 好的，已修改好

**Review by heavengate — DISMISSED** [review]
LG API， need a related RD to check the logic

**Review by nepeplwu — DISMISSED** [review]

**Review by nepeplwu — DISMISSED** [review]

**Review by nepeplwu — DISMISSED** [review]

**Review by nepeplwu — APPROVED** [review]

**Review by heavengate — APPROVED** [review]

**Pull request merged**